From 4bec5d0fa4748d1d54baaba5fa743ff7a1e38e95 Mon Sep 17 00:00:00 2001
From: "Christensen, Taylor" <taylor.christensen@intel.com>
Date: Tue, 3 Nov 2015 16:01:29 -0800
Subject: [PATCH 223/639] PCIe: Basic PCIe Port PM Support

This patch enables the basics of PCIe port PM.
Future patches will fine tune the mechanics for
our platform.
---
 drivers/pci/pcie/portdrv_pci.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/pcie/portdrv_pci.c b/drivers/pci/pcie/portdrv_pci.c
index ce8acc7..244c008 100644
--- a/drivers/pci/pcie/portdrv_pci.c
+++ b/drivers/pci/pcie/portdrv_pci.c
@@ -154,8 +154,9 @@ static int pcie_port_runtime_idle(struct device *dev)
 	 */
 	pci_walk_bus(pdev->subordinate, pci_dev_pme_poll, &pme_poll);
 	/* Delay for a short while to prevent too frequent suspend/resume */
-	if (!pme_poll)
-		pm_schedule_suspend(dev, 10);
+	/* if (!pme_poll) */ /* Disabling pme_poll check for the sake of PM  *
+	                      * on supported devices                         */
+	pm_schedule_suspend(dev, 10);
 	return -EBUSY;
 }
 #else
@@ -217,11 +218,13 @@ static int pcie_portdrv_probe(struct pci_dev *dev,
 	 * it by default.
 	 */
 	dev->d3cold_allowed = false;
+	pm_runtime_put_noidle(&dev->dev);
 	return 0;
 }
 
 static void pcie_portdrv_remove(struct pci_dev *dev)
 {
+	pm_runtime_get_noresume(&dev->dev);
 	pcie_port_device_remove(dev);
 }
 
-- 
2.10.1

